src/metabase/server/middleware/json.clj

```diff
@@ -90,3 +92,45 @@
      (fn respond* [response]
        (respond (wrap-streamed-json-response* opts response)))
      raise)))
+
+;;; +----------------------------------------------------------------------------------------------------------------+
+;;; |                                                JSON Requests                                                   |
+;;; +----------------------------------------------------------------------------------------------------------------+
+
+(defn- parse-json-or-exc [{:keys [body headers] :as req}]
+  (if (and body
+           (some->> (get headers "content-type")
+                    (re-find #"^application/(.+?\+)?json")))
```

Comment:
I'm quite nervous about this part, since we didn't check before. Even if our FE is a good citizen, this could break custom integration.

What do you think about maintaining some backward compatibility - if the header is not present, attempt to parse anyway, but swallow any exceptions?

Reply:
Well, it's in the lib, so it's harder to notice :-) But what I did i just copied the code and then edited it until I was happy with it hehe.

Reply:
Yeah I prefer your code a bit... what is happening to me, I used to love micro helper methods.

Reply:
🧛